diff -urN auth/%%LDAPNAME%%/auth.php auth/%%LDAPNAME%%/auth.php
--- auth/%%LDAPNAME%%/auth.php	2012-12-10 17:02:03.375012706 +0100
+++ auth/%%LDAPNAME%%/auth.php	2012-12-10 17:02:31.139150376 +0100
@@ -83,7 +83,7 @@
 /**
  * LDAP authentication plugin.
  */
-class auth_plugin_ldap extends auth_plugin_base {
+class auth_plugin_%%LDAPNAME%% extends auth_plugin_base {
 
     /**
      * Init plugin config from database settings depending on the plugin auth type.
@@ -145,5 +145,5 @@
      */
     public function __construct() {
-        $this->authtype = 'ldap';
+        $this->authtype = 'ldap2';
         $this->roleauth = 'auth_'.$this->authtype;
         $this->errorlogtag = '[AUTH '.strtoupper($this->authtype).'] ';
diff -urN auth/%%LDAPNAME%%/cli/sync_users.php auth/%%LDAPNAME%%/cli/sync_users.php
--- auth/%%LDAPNAME%%/cli/sync_users.php	2012-12-10 17:02:03.375012706 +0100
+++ auth/%%LDAPNAME%%/cli/sync_users.php	2012-12-10 17:02:31.139150376 +0100
@@ -47,5 +47,5 @@
 define('CLI_SCRIPT', true);
 
-$authtype = 'ldap';
+$authtype = 'ldap2';
 
 require(dirname(dirname(dirname(dirname(__FILE__)))).'/config.php'); // global moodle config file.
diff -urN auth/%%LDAPNAME%%/db/install.php auth/%%LDAPNAME%%/db/install.php
--- auth/%%LDAPNAME%%/db/install.php	2012-11-03 17:53:51.013438496 +0100
+++ auth/%%LDAPNAME%%/db/install.php	2012-12-10 17:02:31.139150376 +0100
@@ -1,6 +1,6 @@
 <?php
 
-function xmldb_auth_ldap_install() {
+function xmldb_auth_%%LDAPNAME%%_install() {
     global $CFG, $DB;
 
 }
diff -urN auth/%%LDAPNAME%%/db/upgrade.php auth/%%LDAPNAME%%/db/upgrade.php
--- auth/%%LDAPNAME%%/db/upgrade.php	2012-11-03 17:53:51.013438496 +0100
+++ auth/%%LDAPNAME%%/db/upgrade.php	2012-12-10 17:02:31.139150376 +0100
@@ -26,9 +26,9 @@
  * @param int $oldversion the version we are upgrading from
  * @return bool result
  */
-function xmldb_auth_ldap_upgrade($oldversion) {
+function xmldb_auth_%%LDAPNAME%%_upgrade($oldversion) {
 
-    $authtype = 'ldap';
+    $authtype = '%%LDAPNAME%%';
     $pluginconfig = 'auth/'.$authtype;
 
     // Moodle v2.5.0 release upgrade line


diff -urN auth/%%LDAPNAME%%/lang/en/auth_%%LDAPNAME%%.php auth/%%LDAPNAME%%/lang/en/auth_%%LDAPNAME%%.php
--- auth/%%LDAPNAME%%/lang/en/auth_%%LDAPNAME%%.php	1970-01-01 01:00:00.000000000 +0100
+++ auth/%%LDAPNAME%%/lang/en/auth_%%LDAPNAME%%.php	2012-12-10 17:05:50.184137388 +0100
@@ -33,12 +33,12 @@
 $string['auth_ldap_bind_settings'] = 'Bind settings';
 $string['auth_ldap_contexts'] = 'List of contexts where users are located. Separate different contexts with \';\'. For example: \'ou=users,o=org; ou=others,o=org\'';
 $string['auth_ldap_contexts_key'] = 'Contexts';
-$string['auth_ldap_create_context'] = 'If you enable user creation with email confirmation, specify the context where users are created. This context should be different from other users to prevent security issues. You don\'t need to add this context to ldap_context-variable, Moodle will search for users from this context automatically.<br /><b>Note!</b> You have to modify the method user_create() in file auth/ldap/auth.php to make user creation work';
+$string['auth_ldap_create_context'] = 'If you enable user creation with email confirmation, specify the context where users are created. This context should be different from other users to prevent security issues. You don\'t need to add this context to ldap_context-variable, Moodle will search for users from this context automatically.<br /><b>Note!</b> You have to modify the method user_create() in file auth/%%LDAPNAME%%/auth.php to make user creation work';
 $string['auth_ldap_create_context_key'] = 'Context for new users';
 $string['auth_ldap_create_error'] = 'Error creating user in LDAP.';
 $string['auth_ldap_creators'] = 'List of groups or contexts whose members are allowed to create new courses. Separate multiple groups with \';\'. Usually something like \'cn=teachers,ou=staff,o=myorg\'';
 $string['auth_ldap_creators_key'] = 'Creators';
-$string['auth_ldapdescription'] = 'This method provides authentication against an external LDAP server.
+$string['auth_%%LDAPNAME%%description'] = 'This method provides authentication against an external LDAP server.
                                   If the given username and password are valid, Moodle creates a new user
                                   entry in its database. This module can read user attributes from LDAP and prefill
                                   wanted fields in Moodle.  For following logins only the username and
@@ -133,7 +133,7 @@
 $string['pagedresultsnotsupp'] = 'LDAP paged results not supported (either your PHP version lacks support or you have configured Moodle to use LDAP protocol version 2)';
 $string['pagesize'] = 'Make sure this value is smaller than your LDAP server result set size limit (the maximum number of entries that can be returned in a single query)';
 $string['pagesize_key'] = 'Page Size';
-$string['pluginname'] = 'LDAP server';
+$string['pluginname'] = 'LDAP %%LDAPNAME%% server';
 $string['pluginnotenabled'] = 'Plugin not enabled!';
 $string['renamingnotallowed'] = 'User renaming not allowed in LDAP';
 $string['rootdseerror'] = 'Error querying rootDSE for Active Directory';
diff -urN auth/%%LDAPNAME%%/ntlmsso_attempt.php auth/%%LDAPNAME%%/ntlmsso_attempt.php
--- auth/%%LDAPNAME%%/ntlmsso_attempt.php	2012-12-10 17:02:03.379012718 +0100
+++ auth/%%LDAPNAME%%/ntlmsso_attempt.php	2012-12-10 17:02:31.139150376 +0100
@@ -2,7 +2,7 @@
 
 require_once(dirname(dirname(dirname(__FILE__))).'/config.php');
 
-$authtype = 'ldap';
+$authtype = '%%LDAPNAME%%';
 
 //HTTPS is required in this page when $CFG->loginhttps enabled
 $PAGE->https_required();
diff -urN auth/%%LDAPNAME%%/ntlmsso_finish.php auth/%%LDAPNAME%%/ntlmsso_finish.php
--- auth/%%LDAPNAME%%/ntlmsso_finish.php	2012-12-10 17:02:03.379012718 +0100
+++ auth/%%LDAPNAME%%/ntlmsso_finish.php	2012-12-10 17:02:31.139150376 +0100
@@ -2,7 +2,7 @@
 
 require_once(dirname(dirname(dirname(__FILE__))).'/config.php');
 
-$authtype = 'ldap';
+$authtype = '%%LDAPNAME%%';
 
 //HTTPS is required in this page when $CFG->loginhttps enabled
 $PAGE->https_required();
diff -urN auth/%%LDAPNAME%%/ntlmsso_magic.php auth/%%LDAPNAME%%/ntlmsso_magic.php
--- auth/%%LDAPNAME%%/ntlmsso_magic.php	2012-12-10 17:02:03.379012718 +0100
+++ auth/%%LDAPNAME%%/ntlmsso_magic.php	2012-12-10 17:02:31.139150376 +0100
@@ -6,7 +6,7 @@
 // of the webserver.
 define('NO_MOODLE_COOKIES', true);
 
-$authtype = 'ldap';
+$authtype = '%%LDAPNAME%%';
 
 require_once(dirname(dirname(dirname(__FILE__))).'/config.php');
 
diff -urN auth/%%LDAPNAME%%/version.php auth/%%LDAPNAME%%/version.php
--- auth/%%LDAPNAME%%/version.php	2012-12-10 17:02:03.379012718 +0100
+++ auth/%%LDAPNAME%%/version.php	2012-12-10 17:02:31.139150376 +0100
@@ -27,3 +27,3 @@
 
-$authtype = 'ldap';
+$authtype = '%%LDAPNAME%%';
 
